tests/test_output.py

```diff
@@ -1176,7 +1174,38 @@ def test_result_list_backup_with_backup_name(self, capsys):
 
         # THEN the console output contains the backup name
         out, _err = capsys.readouterr()
-        assert "%s %s '%s'" % (bi.server_name, bi.backup_id, bi.backup_name) in out
+        assert (
+            "%s %s '%s' - R - " % (bi.server_name, bi.backup_id, bi.backup_name) in out
+        )
+
+    def test_result_list_backup_types(self, capsys):
+        # Rsync backup case (rsync is the default backup method for the test object)
+        rsync_backup_info = build_test_backup_info()
+        self._test_backup_type_helper(rsync_backup_info, "R", capsys)
+
+        # Incremental backup case
+        incremental_backup_info = build_test_backup_info(parent_backup_id="some_id")
+        incremental_backup_info.mode = "postgres"
+        self._test_backup_type_helper(incremental_backup_info, "I", capsys)
+
+        # All others are full backups
+        full_backup_info = build_test_backup_info()
+        full_backup_info.mode = "postgres"
+        self._test_backup_type_helper(full_backup_info, "F", capsys)
+
+    def _test_backup_type_helper(self, backup_info, expected_backup_type, capsys):
+        # Common setup for invoking result_list_backup and checking output
+        console_writer = output.ConsoleOutputWriter()
+        console_writer.init_list_backup(backup_info.server_name, False)
+        console_writer.result_list_backup(backup_info, 12345, 54321, "test status")
+        console_writer.close()
+
+        # Capture and assert output
+        out, _err = capsys.readouterr()
+        assert (
+            f"{backup_info.server_name} {backup_info.backup_id} - {expected_backup_type} -"
+            in out
+        )
```

Comment:
I'd say to remove these unit tests from here, and actually write unit tests for `LocalBackupInfo.backup_type` property in `test_infofile.py`.

What I mean is: in `barman.output` we simply use the returning value of the aforementioned property.

If you want to check that `barman -f console list-backup` uses `LocalBackupInfo.backup_type` as expected, that's surely a valid test. But you need to property and check what `list-backup` does.

For example:

```python
@pytest.mark.parametrize(
    ("backup_type", "expected_value"),
    (
        ("some_type", "S"),
        ("another_type", "A"),
    )
)
def test_result_list_backup_types(...):
    ...
    # Call the following method with a mock on `backup_type` property
    writer.result_list_backup(...)
    # assert that result_list_backup played with "backup_type" and printed "expected_value"